samples/net/mbedtls_sslclient/src/mini_client.c

```diff
@@ -139,28 +141,76 @@ enum exit_codes {
 static int entropy_source(void *data, unsigned char *output, size_t len,
 			  size_t *olen)
 {
-	u32_t seed;
+	struct device *hwrng = data;
+	unsigned char *orig_output = output;
+	size_t orig_len = len;
+
+	/* Use a weak random source if an entropy driver was not found at
+	 * boot.  Please don't do this in an application: set it up properly
+	 * so that a hardware entropy driver is always available.
+	 */
+	if (hwrng == NULL) {
+		u32_t seed;
+
+		seed = sys_rand32_get();
+
+		if (len > sizeof(seed)) {
+			len = sizeof(seed);
+		}
 
-	ARG_UNUSED(data);
+		memcpy(output, &seed, len);
 
-	seed = sys_rand32_get();
+		*olen = len;
 
-	if (len > sizeof(seed)) {
-		len = sizeof(seed);
+		return 0;
 	}
 
-	memcpy(output, &seed, len);
+	/* If an entropy device has been found during boot, use it to gather
+	 * entropy.  This loop is required since `len` is a `size_t`, and the
+	 * equivalent parameter type in `entropy_get_entropy()` is a `u16_t`.
```

Comment:
just curious why is the parameter to entropy_get_entropy() a u16_t anyway?

Reply:
That's how the API has been defined.